contracts/payment/LemonadeEscrowV1.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/finance/PaymentSplitter.sol";
+
+import "./DelegateManager.sol";
+import "./ILemonadeEscrow.sol";
+
+contract LemonadeEscrowV1 is
+    ILemonadeEscrow,
+    Ownable,
+    PaymentSplitter,
+    DelegateManager
+{
+    event GuestDeposit(
+        address guest,
+        uint256 paymentId,
+        address token,
+        uint128 amount
+    );
+    event GuestClaimRefund(address guest, uint256 paymentId);
+    event EscrowClosed();
+    event PaymentCancelled(uint256 paymentId, bool byGuest);
+
+    bool internal _closed;
+    uint16 internal _hostRefundPercent;
+    RefundPolicy[] internal _refundPolicies;
+
+    mapping(uint256 => bool) _paymentCancelled; //-- map paymentId
+    mapping(address => mapping(uint256 => Deposit[])) _deposits; //-- map user -> paymentId
+
+    constructor(
+        address owner,
+        address[] memory delegates,
+        address[] memory payees,
+        uint256[] memory shares,
+        uint16 hostRefundPercent,
+        RefundPolicy[] memory refundPolicies
+    ) PaymentSplitter(payees, shares) DelegateManager(delegates) {
+        require(hostRefundPercent <= 100, "Invalid hostRefundPercent");
+
+        //-- check valid refundPolicies
+        if (refundPolicies.length == 1) {
+            _assertValidRefundPercent(refundPolicies[0]);
+        } else if (refundPolicies.length > 1) {
+            _assertValidRefundPercent(refundPolicies[0]);
+
+            for (uint16 i = 0; i < refundPolicies.length - 1; i++) {
+                require(
+                    refundPolicies[i].timestamp <
```

Comment:
You should refactor this, assign these refundPolicies to intermediate variables, like current and next